### DISCUSSION TITLE
[Feature] Alias by $tag_"some-complicated-name"

### DISCUSSION THREAD

**Discussion by PauloTemenos** [grafana/grafana#37001] [Dashboards, Panels & Visualizations]
Unlike with the [variables ](http://docs.grafana.org/reference/templating/) where we have 2 ways to reference them,
Standard & concise: `$varname`
Alternative failsafe mode: `[[varname]]`

When setting up "Alias by" we can only reference the columns with:
`$tag_ColumnName`
But this reference mechanism cannot reference all column names. Ex, having a dash in the middle like `$tag_column-name` makes the parser look for a column named "column" and then append "-name".

Just like with variables, I propose having a second alternative mechanism to reference column names in the "Alias by" section. The most intuitive option for these situations is to delimit the identifiers with double quotes like `$tag_"funky column-name"`.

_* When querying InfluxDB, its query language does not allow us to select the Tags _(columns)_ "as xyz" like we can for the Fields, so we mandatorily receive the original column name. In Grafana we then become unable to consume those columns in the Alias by section._

**What Grafana version are you using?**
Latest as of 2018-07

**What datasource are you using?**
InfluxDB (N/A?)

**What OS are you running grafana on?**
N/A

**Comment by farthinder** [comment]
I'm also running in to this problem running Grafana v5.3.4  and InfluxDB

**Comment by aocenas** [comment]
@torkelo changed the labels cause I think this is related to general variable interpolation rather than influx data source.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
